From c9edcc55fae70151377cf0c72985e7a76a2dd524 Mon Sep 17 00:00:00 2001
From: Austin Seipp <aseipp@pobox.com>
Date: Sun, 31 May 2015 22:37:57 -0500
Subject: [PATCH] analyzer: make ccc-analyzer use NIX_CFLAGS_COMPILE

Signed-off-by: Austin Seipp <aseipp@pobox.com>
---
 tools/scan-build/ccc-analyzer | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/scan-build/ccc-analyzer b/tools/scan-build/ccc-analyzer
index 4549b29..110df1a 100755
--- a/tools/scan-build/ccc-analyzer
+++ b/tools/scan-build/ccc-analyzer
@@ -230,6 +230,15 @@ sub Analyze {
       push @Args, "-Xclang", "-analyzer-viz-egraph-ubigraph";
     }
 
+
+    # Add Nix flags to analysis
+    if (defined $ENV{'NIX_CFLAGS_COMPILE'}) {
+      my @nixArgs = split(/\s+/, $ENV{'NIX_CFLAGS_COMPILE'});
+      foreach my $nixArg (@nixArgs) {
+          push @Args, $nixArg;
+      }
+    }
+
     my $AnalysisArgs = GetCCArgs("--analyze", \@Args);
     @CmdArgs = @$AnalysisArgs;
   }
-- 
1.9.1

